src/utils/Api.js

```diff
@@ -0,0 +1,102 @@
+class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  // _request(endpoint, options = {}) {
+  //   const finalOptions = {
+  //     headers: this._headers,
+  //     ...options,
+  //   };
+  //   const url = `${this._baseUrl}${endpoint}`;
+  //   return fetch(url, finalOptions).then(this._processResponse);
+  // }
+
+  _processResponse(res){
+    if(res.ok){
+      return res.json();
+    }
+    Promise.reject(`Error: ${res.status}`);
+  }
+
+  getAppInfo() {
+    return Promise.all([this.getInitialCards(), this.getUserInfo()]);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    })
+    .then(this._processResponse)
+  }
+
+  // Implement POST /cards
+
+  addNewCard({ name, link }) {
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify({
+        name,
+        link,
+      }),
+    })
+    .then(this._processResponse)
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    })
+    .then(this._processResponse)
+  }
+
+  editUserInfo({ name, about }) {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      // Send the data in the body as a JSON string.
+      body: JSON.stringify({
+        name,
+        about,
+      }),
+    })
+    .then(this._processResponse)
+  }
+
+  editUserAvatar({ avatar }) {
+    return fetch(`${this._baseUrl}/users/me/avatar`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        avatar
+      }),
+    })
+    .then(this._processResponse)
+  }
+
+
+
+  deleteCard(id) {
+    return fetch(`${this._baseUrl}/cards/${id}`, {
+      method: "DELETE",
+      headers: this._headers,
+    })
```

Comment:
* Please, reuse `.then(this._processResponse)` in every request not to duplicate the `ok` checking


You need to check the response in every request to be able to throw an error, if the response is not `ok`. This error will be caught by the `.catch` in `index.js`


So, in `deleteCard` there should be `.then(this._processResponse);` at the end


```js

  deleteCard(id) {
    return fetch(`${this._baseUrl}/cards/${id}`, {
      method: "DELETE",
      headers: this._headers,
    }).then(this._processResponse);  // here
  }
```